tabled/src/settings/split/mod.rs

```diff
@@ -0,0 +1,174 @@
+//! Split setting is a work in progress
+
+use crate::records::{ExactRecords, Records, Resizable};
+
+use super::TableOption;
+
+#[derive(Debug, Clone, Copy)]
+enum Direction {
+    Column,
+    Row,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum Behavior {
+    Append,
+    Zip,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct Split {
+    direction: Direction,
+    behavior: Behavior,
+    index: usize,
+}
+
+impl Split {
+    pub fn column(index: usize) -> Self {
+        Split {
+            direction: Direction::Column,
+            behavior: Behavior::Zip,
+            index,
+        }
+    }
+
+    pub fn row(index: usize) -> Self {
+        Split {
+            direction: Direction::Row,
+            behavior: Behavior::Zip,
+            index,
+        }
+    }
+
+    pub fn set_behavior(self, behavior: Behavior) -> Self {
+        Self { behavior, ..self }
+    }
+}
+
+impl<RR, D, Cfg> TableOption<RR, D, Cfg> for Split
+where
+    RR: Records + ExactRecords + Resizable,
+{
+    fn change(&mut self, records: &mut RR, _: &mut Cfg, _: &mut D) {
+        let Split {
+            direction,
+            behavior,
+            index,
+        } = &self;
+        let before_rows = records.count_rows();
+        let before_columns = records.count_columns();
+
+        match (direction, behavior) {
+            (Direction::Column, Behavior::Append) => {
+                let sections_per_row =
+                    before_columns / index + { usize::from(before_columns % index != 0) };
+
+                for section in 1..sections_per_row {
+                    for reference_row in 0..before_rows {
+                        let target_row = reference_row + section * before_rows;
+                        records.push_row();
+
+                        for target_column in 0..*index {
+                            let reference_column = target_column + section * index;
+
+                            if reference_column < before_columns {
+                                records.swap(
+                                    (target_row, target_column),
+                                    (reference_row, reference_column),
+                                );
+                            }
+                        }
+                    }
+                }
+
+                for column in (*index..before_columns).rev() {
+                    records.remove_column(column)
+                }
```

Comment:
Maybe move the code in the branches to the functions?
And leave only match in the function.

Likely wont be necessary then?

https://github.com/zhiburt/tabled/blob/0710745eb4c35e1b3018de2fffc62b0c04bd8af0/tabled/src/settings/split/mod.rs#L53-L57

Reply:
https://github.com/zhiburt/tabled/pull/299/files#diff-881689aca954c8702c2bde0c3bf79fb1240972e785c9e672358dcc1407c05e33R80-R192

I'm on the fence about this change. I think I've made a lot more code 😅

Reply:
I guess it's better though (yes.... more lines but seems more clear).
Made a few refactorings 39b628b